src/components/screens/vacation-requests-screen.tsx

```diff
@@ -13,31 +13,37 @@ import { errorAtom } from "../../atoms/error";
 import { GridRowId } from "@mui/x-data-grid";
 import { VacationData } from "../../types";
 import strings from "../../localization/strings";
-import { vacationRequestsAtom, vacationRequestStatusesAtom } from "../../atoms/vacation";
+import {
+  allVacationRequestsAtom,
+  allVacationRequestStatusesAtom,
+  vacationRequestsAtom,
+  vacationRequestStatusesAtom
+} from "../../atoms/vacation";
+import UserRoleUtils from "../../utils/user-role-utils";
+import { Link } from "react-router-dom";
+import { KeyboardReturn } from "@mui/icons-material";
 
 /**
  * Vacation requests screen
  */
 const VacationRequestsScreen = () => {
+  const adminMode = UserRoleUtils.adminMode();
   const { vacationRequestsApi, vacationRequestStatusApi } = useApi();
   const userProfile = useAtomValue(userProfileAtom);
   const setError = useSetAtom(errorAtom);
-  const [vacationRequests, setVacationRequests] = useAtom(vacationRequestsAtom);
-  const [vacationRequestStatuses, setVacationRequestStatuses] = useState<VacationRequestStatus[]>(
-    []
+  const [vacationRequests, setVacationRequests] = useAtom(
+    adminMode ? allVacationRequestsAtom : vacationRequestsAtom
+  );
+  const [latestVacationRequestStatuses, setLatestVacationRequestStatuses] = useAtom(
+    adminMode ? allVacationRequestStatusesAtom : vacationRequestStatusesAtom
```

Comment:
Vacations are fetched in the home-screen (because vacations-card needs them) and stored in the atom so that the vacations don't have to be fetched again when navigating to the vacation-requests-screen. 

Using atom for vacations also saves us from tons of prop drilling in the vacation-requests-table component and its child components.

latestVacationRequestStatuses state name is to make a distinction between unfiltered and filtered vacation requests. Statuses need to be filtered because there can be multiple of them for one vacation request even though only one is needed to display in the table. The latest status is chosen for one vacation request.